PWGCF/FemtoDream/Core/femtoDreamCollisionSelection.h

```diff
@@ -180,9 +185,65 @@ class FemtoDreamCollisionSelection
   /// \param tracks All tracks
   /// \return value of the sphericity of the event
   template <typename T1, typename T2>
-  float computeSphericity(T1 const& /*col*/, T2 const& /*tracks*/)
+  float computeSphericity(T1 const& col, T2 const& tracks)
   {
-    return 2.f;
+    double ptTot = 0.;
+    double s00 = 0.; // elements of the sphericity matrix taken form EPJC72:2124
+    double s01 = 0.;
+    double s10 = 0.;
+    double s11 = 0.;
+
+    int numOfTracks = col.numContrib();
+    if (numOfTracks < 3)
+      return -9999.;
+
+    for (auto& track : tracks) {
+      double pt = track.pt();
+      double eta = track.eta();
+      double px = track.px();
+      double py = track.py();
+      if (TMath::Abs(pt) < /*lowerPtbound*/ 0.5 || TMath::Abs(eta) > 0.8) {
+        continue;
+      }
+
+      ptTot += pt;
+
+      s00 += px * px / pt;
+      s01 += px * py / pt;
+      s10 = s01;
```

Comment:
you never use this variable later and also don't properly normalize to total pT, it is better to write it as a comment in s01 line that one of the other matrix elements you need is s10 but it is equal to s01 so you use them as quivalent.